js.go

```diff
@@ -2144,7 +2144,8 @@ type streamNamesResponse struct {
 	Streams []string `json:"streams"`
 }
 
-func (js *js) lookupStreamBySubject(subj string) (string, error) {
+// StreamNameBySubject returns a stream name that matches the subject.
+func (js *js) StreamNameBySubject(subj string) (string, error) {
```

Comment:
Please add a test case for subject without wildcard (e.g. `test.A`) and split the test to several test cases (preferably as table-driven unit test, as this is a good use case for this approach).